contracts/BasicPolyptychRandomizerV0.sol

```diff
@@ -0,0 +1,133 @@
+ * it enables the copying of token hashes from an original token (the first panel of a polyptych)
+ * to subsequent diptych, triptych, or polyptych panels.
+ *
+ * Requirements to use the polyptych randomizer:
+ * - Core contract must be an instance of `GenArt721CoreV3_Engine`
+ * - The initial project used as the first panel must be minted with a completely random token hash
+ * - An `MinterPolyptychV0` contract or similar must be deployed and configured as the `hashSeedSetterContract`
+ * - The `projectDuplicateHashSeedLimit` should be configured for each project
+ *
+ * Once the contract is configured, it will enable newly-minted tokens to use a copy of the hash
+ * seed from a previously-minted token.
+ */
+contract BasicPolyptychRandomizerV0 is IBasicPolyptychRandomizerV0, Ownable {
+    // The core contract that may interact with this randomizer contract.
+    IGenArt721CoreContractV3 public genArt721Core;
+
+    // Used to obtain the project ID from the token ID
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    // Mapping of token IDs to hash seeds
+    mapping(uint256 => bytes12) public polyptychHashSeeds;
+
+    // Mapping of the number of times a hash seed has been used
+    mapping(bytes12 => uint) public hashSeedUsageCounter;
+
+    // Configures the maximum number of hash seed copies that can exist per project (default zero)
+    mapping(uint256 => uint) public projectDuplicateHashSeedLimit;
+
+    // The contract allowed to assign hash seeds (e.g. PolyptychMinterV0)
+    address public hashSeedSetterContract;
+
+    function assignCoreAndRenounce(address _genArt721Core) external onlyOwner {
+        renounceOwnership();
+        genArt721Core = IGenArt721CoreContractV3(_genArt721Core);
+    }
+
+    // modifier to restrict access to only AdminACL allowed calls
+    // @dev defers which ACL contract is used to the core contract
+    modifier onlyCoreAdminACL(bytes4 _selector) {
+        require(
+            genArt721Core.adminACLAllowed(msg.sender, address(this), _selector),
+            "Only Core AdminACL allowed"
+        );
+        _;
+    }
+
+    // Allows the owner of the core contract to set the minter that is allowed to assign hash seeds
+    function setHashSeedSetterContract(
+        address _contractAddress
+    ) external onlyCoreAdminACL(this.setHashSeedSetterContract.selector) {
+        hashSeedSetterContract = _contractAddress;
+        emit HashSeedSetterUpdated(_contractAddress);
+    }
+
+    /**
+     * @notice Allows the owner of the core contract to configure a project as a polyptych
+     * @param _projectId - The ID of the project that is a polyptych panel (second, third, etc.)
+     * @param _maxDuplicates - The number of previous panels in the polyptych
+     */
+    function setprojectDuplicateHashSeedLimit(
+        uint256 _projectId,
+        uint _maxDuplicates
+    )
+        external
+        onlyCoreAdminACL(this.setprojectDuplicateHashSeedLimit.selector)
```

Comment:
This function might be better as artist-only instead of admin?

Reply:
++; also should use `uint256` here rather than `uint` I think, per rationale discussed above.